wordpress/plugins/kredeum-nfts/admin/settings/class-settings.php

```diff
@@ -21,20 +23,37 @@ class Settings {
 	 *
 	 * @var string $slug Settings slug
 	 */
-	private $slug = 'ipfs_settings';
+	private $slug = STORAGE . '_settings';
 
 	/**
 	 * Constructor
 	 */
 	public function __construct() {
 		// Hook into the admin menu.
-		add_action( 'admin_menu', array( $this, 'page_create' ) );
+		if ( STORAGE === 'ipfs' ) {
+			add_action( 'admin_menu', array( $this, 'page_create' ) );
+		} else {
+			add_action( 'admin_menu', array( $this, 'menu_create' ) );
+		}
```

Comment:
specific condition to ipfs.
tried to resolve the problem with /admin/swarm/index.php file but didn't succeed.